pkg/lightning/restore/restore.go

```diff
@@ -911,7 +922,14 @@ func (rc *Controller) saveStatusCheckpoint(tableName string, engineID int32, err
 		metric.RecordEngineCount(statusIfSucceed.MetricName(), err)
 	}
 
-	rc.saveCpCh <- saveCp{tableName: tableName, merger: merger}
+	waitCh := make(chan error, 1)
+	rc.saveCpCh <- saveCp{tableName: tableName, merger: merger, waitCh: waitCh}
+
+	if err = <-waitCh; err != nil {
```

Comment:
will this cause Lightning unable to exit by Ctrl+C since we're blocked here

Reply:
It might be possible if `checkpoints.DB.Update` is blocked too long.  I will consider adding a ctx parameter to this method later.